### PR TITLE
development: move/simplify dev server setup and execution (bug 1697244)

### PR DESCRIPTION
**Pull request by zzzeid**
- move logging and setting of config params to own method
- create method to fetch version info
- instead of storing path to version info, store version info directly
- update Dockerfile to be based off of latest python (slim) images
    - move system packages to own requirements file
- create docker-compose file for CircleCI and update CircleCI config
- minor change to setup.py

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,93 +10,81 @@
 # DOCKER_PASS
 
 version: 2
-
-# This is not part of circleci, but it lets us not repeat ourselves.
-STUFF:
```
Comment (mitchhentges) [inline .circleci/config.yml:15]: :heart_eyes: thanks for removing `STUFF` haha :laughing:

File: .circleci/config.yml [hunk 2]
```diff
@@ -105,14 +93,8 @@ workflows:
...
-      - build:
-          filters: *main_filters
-          requires:
-            - lint
-            - test
+      - full:
+          filters:
+            branches:
+              ignore:
+                - /\//
```
Comment (mitchhentges) [inline .circleci/config.yml:100]: Would you mind re-adding the comment about slashes? It's unclear why these are being ignored now.

File: .circleci/config.yml [hunk 3]
```diff
@@ -10,93 +10,81 @@
...
-      - run: apk add gcc linux-headers libc-dev
-      - run: pip install black==19.10b0 && pip install flake8==3.7.9
-      - run: black --diff .
-      - run: flake8 .
-  test:
-    working_directory: /app
-    docker:
-      - *python_image
+      - image: circleci/python
+        user: root
```
Comment (mitchhentges) [inline .circleci/config.yml:17]: Do you need the `circleci/python` image if you aren't doing any Python commands in the Circle CI shell? Looks like all the Python work is happening within the `docker/Dockerfile-prod/` image

File: .circleci/config.yml [hunk 4]
```diff
@@ -10,93 +10,81 @@
...
+                \"build\": \"$CIRCLE_BUILD_URL\"
+              }
+            " > version.json
+            mkdir /tmp/artifacts && cp version.json /tmp/artifacts/version.js
 
+      - run:
+          name: Build the image
+          command: |
+            docker build --pull \
+              -f docker/Dockerfile-prod\
```
Comment (mitchhentges) [inline .circleci/config.yml:38]: What does the backslash at the end here do?

File: docker/Dockerfile-dev [hunk 5]
```diff
@@ -1,34 +1,25 @@
-FROM python:3.8-alpine
+FROM python:3.9-slim
```
Comment (mitchhentges) [inline docker/Dockerfile-dev:1]: Nice

File: docker/Dockerfile-prod [hunk 6]
```diff
@@ -2,61 +2,54 @@
...
-        \"version\": \"$CIRCLE_SHA1\", \
-        \"source\": \"https://github.com/mozilla-conduit/lando-ui\", \
-        \"build\": \"$CIRCLE_BUILD_URL\" \
-    } \
-    " > /version.json
-
-RUN apk add --update ruby && rm -rf /var/cache/apk/*
 RUN gem install sass --no-document -v 3.4.25
 
+# TODO get rid of the below.
```
Comment (mitchhentges) [inline docker/Dockerfile-prod:21]: Is this work for the future, or for this PR?

File: setup.py [hunk 7]
```diff
@@ -25,17 +25,11 @@
         "Intended Audience :: Developers",
         "Topic :: Software Development :: Build Tools",
         "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
-        "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
```
Comment (mitchhentges) [inline setup.py:28]: Nice.

File: system_requirements.txt [hunk 8]
```diff
@@ -0,0 +1,7 @@
+ gcc
+ git
+ libc-dev
+ musl-dev
+ libffi-dev
+ nodejs
+ ruby
```
Comment (mitchhentges) [inline system_requirements.txt:7]: Should we pin these? Maybe work for a future PR.

**Review by mitchhentges — APPROVED** [review]
LGTM. There's a lot of config shuffling, and my `docker-compose`-fu isn't at full juice atm, but I trust that this has been tested and is :100: Nice work!

File: .circleci/config.yml (hunk 3 above)
Comment (zzzeid) [reply .circleci/config.yml:17]: Correct. Will switch this to a more bare bones image.

File: docker/Dockerfile-prod (hunk 6 above)
Comment (zzzeid) [reply docker/Dockerfile-prod:21]: This is future work. the `auth` module needs to be refactored a little before these can be removed.

File: .circleci/config.yml (hunk 4 above)
Comment (mitchhentges) [reply .circleci/config.yml:38]: Oh, I see. I wasn't sure if it was somehow part of the path.
Would you mind adding a space between `Dockerfile-prod` and `\`?

**Review by mitchhentges — APPROVED** [review]
Nice, thanks! :tada:

**Pull request merged**